be2-scala/src/main/scala/ch/epfl/pop/decentralized/Monitor.scala

```diff
@@ -25,18 +25,23 @@ final case class Monitor(
   private val periodicHbKey = 0
   private val singleHbKey = 1
 
+  // State of connected servers
+  private var someServerConnected = false
+
   // Monitor is self-contained,
   // To that end it doesn't know the ref of the connectionMediator
   private var connectionMediatorRef = ActorRef.noSender
 
   override def receive: Receive = LoggingReceive {
 
     case Monitor.AtLeastOneServerConnected =>
+      someServerConnected = true
       connectionMediatorRef = sender()
       if (!timers.isTimerActive(periodicHbKey))
```

Comment:
You could use `someServerConnected` for this condition too

Reply:
Not sure, the logic here is that I want the timer to be set again only when it wasn't, regardless of server being connected or not. I just want to avoid the very narrow corner case where for some reason `Monitor` receives repetitively `AtLeastOneServerConnected` and have the timer reset every time.

Reply:
You can do the following to use `someServerConnected`:
```scala
connectionMediatorRef = sender()
if (!someServerConnected) {
  timers.startTimerWithFixedDelay(periodicHbKey, TriggerHeartbeat, heartbeatRate)
  someServerConnected = true
}
```